lib/do_profile/profilecommon.js

```diff
@@ -135,6 +137,213 @@ function setCurrentProfile(opts, cb) {
     });
 }
 
+/**
+ * Generate a cmon certificate/key pair for the given profile. This means
+ * checking the cloudapi has a cmon service (ListServices), finding the user's
+ * SSH *private* key, and creating the client certs that will be used to talk
+ * to the Triton CMON service.
+ *
+ * @param {Object} opts: Required.
+ *      - {Object} cli: Requried. The Triton CLI object.
+ *      - {String} name: Required. The profile name.
+ *      - {Boolean} yes: Optional. Boolean indicating if confirmation prompts
+ *        should be skipped, assuming a "yes" answer.
+ *      - {Number} lifetime: Optional. Number of days to make the CMON
+ *        certificate valid for. Defaults to 3650 (10 years).
+ */
+function profileCmonCertgen(opts, cb) {
+    assert.object(opts.cli, 'opts.cli');
+    assert.string(opts.name, 'opts.name');
+    assert.optionalBool(opts.yes, 'opts.yes');
+    assert.optionalNumber(opts.lifetime, 'opts.lifetime');
+    assert.func(cb, 'cb');
+
+    /* Default to a 10 year certificate. */
+    if (!opts.lifetime)
+        opts.lifetime = DEFAULT_LIFETIME;
+
+
+    var cli = opts.cli;
+    var tritonapi = cli.tritonapiFromProfileName({profileName: opts.name});
+
+    var yes = Boolean(opts.yes);
+    var log = cli.log;
+
+    var profile = tritonapi.profile;
+    var cmonHost;
+
+    var agent = new sshpk_agent.Client();
+    var sslCert;
+    var sslKey;
+
+    vasync.pipeline({arg: {tritonapi: tritonapi}, funcs: [
+        function cmonKeyNotice(arg, next) {
+            console.log(wordwrap('Note: CMON uses authentication via ' +
+                'client TLS certiificates.\n'));
+            console.log(wordwrap('This action will create a ' +
+                'fresh private key to be written unencrypted to disk ' +
+                'in the current working directory. Copy these files to ' +
+                'your cmon client (whether Prometheus, or something ' +
+                'else).\n'));
+            console.log(wordwrap('This key will be usable only for CMON. ' +
+                'If your SSH key is removed from your account, this CMON ' +
+                'key will no longer work.\n'));
+            if (yes) {
+                next();
+                return;
+            } else {
+                console.log(wordwrap('If you do not specifically want to use ' +
```

Comment:
```suggestion
                console.log(wordwrap('If you do not want to use ' +
```

Reply:
This also is for consistency with the Docker set up. I'm biased toward consistency, so I can change them both or leave both as-is.

Reply:
Your call :)

Reply:
I prefer this one as-is.